ResearchKit/ActiveTasks/ORKTimedWalkStepViewController.m

```diff
@@ -0,0 +1,127 @@
+ list of conditions and the following disclaimer.
+ 
+ 2.  Redistributions in binary form must reproduce the above copyright notice,
+ this list of conditions and the following disclaimer in the documentation and/or
+ other materials provided with the distribution.
+ 
+ 3.  Neither the name of the copyright holder(s) nor the names of any contributors
+ may be used to endorse or promote products derived from this software without
+ specific prior written permission. No license is granted to the trademarks of
+ the copyright holders even if such marks are included in this software.
+ 
+ THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE
+ FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+
+#import "ORKTimedWalkStepViewController.h"
+#import "ORKTimedWalkContentView.h"
+#import "ORKTimedWalkStep.h"
+#import "ORKPedometerRecorder.h"
+#import "ORKActiveStepViewController_Internal.h"
+#import "ORKNavigationContainerView_Internal.h"
+#import "ORKStepViewController_Internal.h"
+#import "ORKActiveStepView.h"
+
+
+double const kDistanceInMetersTrackingThreshold = 100.0;
+
+
+@interface ORKTimedWalkStepViewController ()
+
+@property (nonatomic, strong) NSMutableArray *samples;
+@property (nonatomic, strong) ORKTimedWalkContentView *timedWalkContentView;
+@property (nonatomic, assign) NSTimeInterval trialDuration;
+
+@end
+
+@implementation ORKTimedWalkStepViewController
+
+- (instancetype)initWithStep:(ORKStep *)step {
+    self = [super initWithStep:step];
+    
+    if (self) {
+        self.suspendIfInactive = YES;
+    }
+    
+    return self;
+}
+
+- (ORKTimedWalkStep *)timedWalkStep {
+    return (ORKTimedWalkStep *)self.step;
+}
+
+- (void)initializeInternalButtonItems {
+    [super initializeInternalButtonItems];
+    
+    self.internalDoneButtonItem = nil;
+    self.continueButtonTitle = ORKLocalizedString(@"BUTTON_DONE", nil);
+}
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    
+    self.timedWalkContentView = [ORKTimedWalkContentView new];
+    self.timedWalkContentView.image = [self timedWalkStep].image;
+    self.activeStepView.activeCustomView = self.timedWalkContentView;
+    self.activeStepView.stepViewFillsAvailableSpace = YES;
+    self.activeStepView.continueSkipContainer.continueEnabled = YES;
+    
+    self.timerUpdateInterval = 0.1f;
+}
+
+- (void)finish {
```

Comment:
Unfortunately not, cause I need to automatically go forward once time is up.
And I cannot set shouldContinueOnFinish to YES otherwise I won't see the continue button.